kbase/narrative
# DATAUP-330 Add test docs

# Description of PR purpose/changes

Add some documentation around front end unit testing and the mocks library, and rearrange the doc files so there's a testing directory. Also a couple minor changes to that library.

# Jira Ticket / Issue #
https://kbase-jira.atlassian.net/browse/DATAUP-330
- [x] Added the Jira Ticket to the title of the PR

# Testing Instructions
* Details for how to test the PR:
Read the docs and see if they fit the process we've been using. There should be no change to unit/integration tests here.

# Dev Checklist:
No code changes of any note!
